### PR TITLE
Setup Catalog and OpenTelemetry Collector

### PR DESCRIPTION
**Pull request by delvh**
Fixes https://frontend.gropius.duckdns.org/components/2571e2de-a0e7-4017-b721-2fd3692ad5d6/issues/f501e77c-81b4-4375-a2fb-0e18d18c8c53

Addresses https://frontend.gropius.duckdns.org/components/2571e2de-a0e7-4017-b721-2fd3692ad5d6/issues/34f9d95c-64f6-4888-bf1d-16e4cf8cd639 (but perhaps not fixes completely)

## DoD

- [x] Requirements of the issue are met
- [x] `terraform apply` works
- [x] `kubectl get pod --namespace misarch` afterwards shows all pods up and running

### CODE REVIEW

File: misarch-catalog.tf [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            name  = "SPRING_R2DBC_URL"
+            value = "r2dbc:postgresql://${local.catalog_db_url}/${var.MISARCH_DB_DATABASE}"
+          }
+
+          env {
+            name  = "SPRING_FLYWAY_URL"
+            value = "jdbc:postgresql://${local.catalog_db_url}/${var.MISARCH_DB_DATABASE}"
+          }
+
+          env {
```
Comment (nk-coding) [inline misarch-catalog.tf:58]: this should not be required, as you set all the properties which this profile provides in dev mode

File: variables-misc.tf [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  sensitive = true
+  type      = string
+}
+
+variable "KUBERNETES_NAMESPACE" {
+  type    = string
+  default = "misarch"
+}
+
+variable "MISARCH_DB_USER" {
```
Comment (nk-coding) [inline variables-misc.tf:11]: does this stuff map well to all our used databases?

File: variables-urls.tf [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  user_db_service_name         = "user-db"
+  wishlist_db_service_name     = "wishlist-db"
+
+  otel_collector_service_name = "otel-collector"
+
+  db_port = 5432
+  otel_collector_port = 4317
+}
+locals {
+  // The Postgresql HA Helm chart always appends '-postgresql', so we would need to add it to the URL too, if we switched to it
```
Comment (nk-coding) [inline variables-urls.tf:30]: do we ever plan to switch to the HA chart?

Comment (delvh) [reply variables-urls.tf:30]: I originally tried to use the HA chart, but it made too many problems.
However, there is a second reason why we need it: If we ever decide to switch from a deployment to a statefulset, we will need this variables again.
As such, I simply left the already written code in as it does not decrease maintainability and does not affect performance.

File: variables-misc.tf (hunk 2 above)
Comment (delvh) [reply variables-misc.tf:11]: What do you mean?
I'm redefining all DBs myself in this PR, that's why it "maps well" to our DB.
So yes, the DB can be queried successfully.
Or what do you mean with `maps well`?

Comment (delvh) [reply variables-misc.tf:11]: MongoDB has users, DBs, and Helm Chart Authentication, so it should work

**Review by nk-coding — APPROVED** [review]

**Pull request merged**